backend/review/import_utils/import_to_db.py

```diff
@@ -0,0 +1,312 @@
+import re
+import sys
+import uuid
+
+from django.contrib.auth import get_user_model
+from tqdm import tqdm
+
+from courses.models import Instructor
+from courses.util import (
+    get_or_create_course,
+    get_or_create_course_and_section,
+    separate_course_code,
+)
+from review.models import Review, ReviewBit
+
+
+"""
+PCR SQL DUMP IMPORT FUNCTIONS
+
+The functions in this file represent the "back-end" of the PCR data import.
+Dictionaries representing rows of PCR data come in and are written to the proper
+database objects. Care is taken not to duplicate rows by using get_or_create().
+
+Helper functions for importing instructors, sections, courses and reviews take in
+specific values extracted from dictionaries. This allows for re-use between import
+passes (for example, the SUMMARY file and the CROSSLISTINGS file), as well as allowing for
+more granular unit tests.
+"""
+
+
+"""
+Review Bits have different labels in the summary table and the rating table.
+This tuple keeps track of the association between the two, along with an
+intermediate, general label that we use internally.
+"""
+REVIEW_BIT_LABEL = (
+    ("RINSTRUCTORQUALITY", "Instructor Quality", "instructor_quality"),
+    ("RCOURSEQUALITY", "Course Quality", "course_quality"),
+    ("RCOMMABILITY", "Comm. Ability", "communication_ability"),
+    ("RSTIMULATEINTEREST", "Stimulate Ability", "stimulate_interest"),
+    ("RINSTRUCTORACCESS", "Instructor Access", "instructor_access"),
+    ("RDIFFICULTY", "Difficulty", "difficulty"),
+    ("RWORKREQUIRED", "Work Required", "work_required"),
+    ("RTAQUALITY", "TA Quality", "ta_quality"),
+    ("RREADINGSVALUE", "Readings Value", "readings_value"),
+    ("RAMOUNTLEARNED", "Amount Learned", "amount_learned"),
+    ("RRECOMMENDMAJOR", "Recommend Major", "recommend_major"),
+    ("RRECOMMENDNONMAJOR", "Recommend Non-Major", "recommend_nonmajor"),
+    ("RABILITIESCHALLENGED", "Abilities Challenged", "abilities_challenged"),
+    ("RCLASSPACE", "Class Pace", "class_pace"),
+    ("RINSTRUCTOREFFECTIVE", "Instructor Effectiveness", "instructor_effective"),
+    ("RNATIVEABILITY", "Native Ability", "native_ability"),
+)
+
+# Maps column name from SUMMARY sql tables to common slug.
+COLUMN_TO_SLUG = dict([(x[0], x[2]) for x in REVIEW_BIT_LABEL])
+# Maps "context" value from RATING table to common slug.
+CONTEXT_TO_SLUG = dict([(x[1], x[2]) for x in REVIEW_BIT_LABEL])
+
+
+def titleize(name):
+    """
+    Titleize a course name or instructor, taking into account exceptions such as II.
```

Comment:
Would it be possible to explicitly state the exceptions? Regex isn't super readable.

Reply:
I copy/pasted this from PCR's code lmao but I'll try

Reply:
Maybe it's a good idea to write tests for this function directly too.